src/goto-symex/symex_assign.cpp

```diff
@@ -269,31 +291,38 @@ static assignmentt shift_indexed_access_to_lhs(
         if(byte_extract.id() == ID_index)
         {
           index_exprt &idx = to_index_expr(byte_extract);
-
-#ifdef USE_UPDATE
-          update_exprt new_rhs{idx.array(), exprt{}, ssa_rhs};
-          new_rhs.designator().push_back(index_designatort{idx.index()});
-#else
-          with_exprt new_rhs{idx.array(), idx.index(), ssa_rhs};
-#endif
-
-          ssa_rhs = new_rhs;
+          ssa_rhs = [&]() -> exprt {
+            if(use_update)
+            {
+              update_exprt new_rhs{idx.array(), exprt{}, ssa_rhs};
+              new_rhs.designator().push_back(index_designatort{idx.index()});
+              return std::move(new_rhs);
+            }
+            else
+              return with_exprt{idx.array(), idx.index(), ssa_rhs};
+          }();
```

Comment:
I don't get the point of this lambda function?!

Reply:
This is to make clear that this part of the code is just to find the new value of ssa_rhs. The alternative is to have directly an `if` statement and we wouldn't know what it's mutating without looking at the code inside the branches.